packages/lesswrong/server/vulcan-lib/mutators.ts

```diff
@@ -472,10 +471,10 @@ export const updateMutator: UpdateMutator = async <N extends CollectionNameStrin
   if (!isEmpty(modifier)) {
     // update document
     logger('updating document')
-    await Connectors.updateOne(collection, selector, modifier, { removeEmptyStrings: false });
+    await collection.rawUpdateOne(convertDocumentIdToIdInSelector(selector), modifier);
```

Comment:
I've at least checked that nothing using `useUpdate` on the client is passing in a selector with a `documentId` field, but this still feels like a scary sort of operation to perform, since I think if we do end up performing this transformation it could bypass permissions checks on who's allowed to update documents?  Maybe we also want to modify the `check` in default update mutation to do a permissions check based on the converted selector?  (Not sure that ever ends up mattering, either.)

Reply:
Oh, I guess we were already doing this inside of `Connectors.updateOne`, so nothing gets worse?  If so, ignore the above

Reply:
Right, this deranged functionality has been moved out of Connectors, but should apply to exactly the set of places it did before. The main change in practice is that now it has a console log, so we can check whether it's getting triggered anywhere and remove it if it isn't.